### DISCUSSION TITLE
Testing AES hardware acceleration support using the latest openssl versions 3.1.1 or higher

### DISCUSSION THREAD

**Discussion by appapurapu** [openssl/openssl#21712] [General Discussion]
We have c++ based cross platform applications(Linux, Windows, Android, web, etc) linked with openssl static libraries.
I am trying to test the hardware acceleration support using openssl EVP_* apis, but I am unable to find one which finds it.
In earlier releases, we can see couple of APIs like OPENSSL_ia32cap_loc() or CRYPTO_get_hw_acceleration_enabled() or EVP_has_aes_hardware(), etc. exists but not in the latest releases.
Are there any openssl public APIs exists to determine the AES hardware acceleration support? If not, what is the best approach for the c++ applications (with static openssl libraries) to test aes hardware acceleration support ?
is the latest OpenSSL 3.1.1 or higher versions uses the CPUID instruction to determine AES hardware acceleration support?

**Comment by t8m** [comment]
Duplicate of #21710
